modules/drenseq.nf

```diff
@@ -104,162 +86,307 @@ process BowtieAlign {
+    path 'nlr_bait_regions.bed'
     script:
     """
-    freebayes \
-      -f ${reference} \
-      -t ${bed} \
-      --min-alternate-count 2 \
-      --min-alternate-fraction 0.05 \
-      --ploidy 4 \
-      -m 0 \
-      -v variants.vcf \
-      --legacy-gls ${bam}
-
-    bcftools sort -o variants.sorted.vcf variants.vcf
-
-    bgzip -c variants.sorted.vcf > ${bam.baseName}.vcf.gz
-    tabix -p vcf ${bam.baseName}.vcf.gz
+    bedtools intersect -a $trimmed_bed -b $bait_regions > nlr_bait_regions.bed
     """
 }
 
-process MergeVCFs {
-    //conda conda_env
-    container 'swiftseal/drenseq:latest'
+process BaitBlastCheck {
+    container 'docker://quay.io/biocontainers/python:3.12'
+    scratch true
     cpus 1
-    memory { 8.GB * task.attempt }
+    memory { 1.GB * task.attempt }
     errorStrategy { task.exitStatus == 137 ? 'retry' : 'finish' }
     maxRetries 3
     time '1h'
     input:
-    path vcf_files
-    path reference
-    path bed
+    path nlr_bait_regions_bed
+    path reference_headers
     output:
-    path 'merged.vcf'
+    path 'passed_genes.txt'
+    path 'missing_genes.txt'
+    publishDir 'results/diagnostics', mode: 'copy'
     script:
     """
-    VCF_FILES=\$(ls *.vcf.gz)
-    bcftools merge -o merged.vcf \$VCF_FILES
+    #!/usr/bin/env python3
+
+    bed_nlr = set()
+
+    with open("$nlr_bait_regions_bed") as bed:
+        for line in bed:
+            bed_nlr.add(line.strip().split()[3])
+    
+    with open("$reference_headers") as headers:
+        next(headers) # skip the header
+        for line in headers:
+            with open("missing_genes.txt", "w") as missed:
+                nlr = line.strip()
+                if nlr not in bed_nlr:
+                    string_to_write = nlr + " not found in bed file"
+                    print(string_to_write, file = missed)
+        with open("missing_genes.txt", "w") as missed:
+            print("\\n", file = missed)
+    
+    with open("passed_genes.txt", "w") as passed:
+        for nlr in bed_nlr:
+            print(nlr, file = passed)
     """
 }
 
-process CoverageMatrix{
-    //container 'https://depot.galaxyproject.org/singularity/r-tidyverse:1.2.1'
+process BedtoolsCoverage {
+    container 'docker://quay.io/biocontainers/bedtools:2.31.1--hf5e1c6e_2'
+    scratch true
     cpus 1
     memory { 1.GB * task.attempt }
     errorStrategy { task.exitStatus == 137 ? 'retry' : 'finish' }
     maxRetries 3
     time '1h'
     input:
-    path txt
+    path bait_regions_bed
+    tuple val(sample), path(bam)
+    path passed_genes
     output:
-    path 'merged.csv'
+    tuple val(sample), path('coverage.txt')
     script:
     """
-    #!/usr/bin/env Rscript
-    # Could use data.table, but I'm lazy :-)
-    library(tidyverse)
+    bedtools coverage -d \
+        -a $bait_regions_bed \
+        -b $bam \
+        > coverage.txt
+    """
+}
 
-    EXTENSION <- ".coverage.txt"
+process PerGeneCoverage {
```

Comment:
diabolical bash

Reply:
but fine lol

Reply:
actually what was wrong with the R script :'(

Reply:
I didn't see how it was going to produce the same output with the same checks as the old workflow, so just went back to what we had in HISS